### PR TITLE
Vehicle Framework Armor

### PR DESCRIPTION
**Pull request by perkinslr**
## Changes

Vehicle's handle hit location and armor properly

## Testing

Check tests you have performed:
- [x] Compiles without warnings
- [x] Game runs without errors
- [ ] (For compatibility patches) ...with and without patched mod loaded
- [x] Playtested a colony (specify how long)

### CODE REVIEW

File: Source/VehiclesCompat/VehiclesCompat/VehicleArmorPatch.cs [hunk 1]
```diff
@@ -0,0 +1,344 @@
...
+namespace CombatExtended.Compatibility.VehiclesCompat
+{
+    [HarmonyPatch(typeof(VehicleStatHandler),
+		  nameof(VehicleStatHandler.ApplyDamageToComponent))]
+    public static class VehicleArmorPatch
+    {
+	
+	public static bool Prefix(VehicleStatHandler __instance, ref DamageInfo dinfo, IntVec2 hitCell, StringBuilder report)
+	{
+	    if (Controller.settings.patchArmorDamage)
```
Comment (trikanyo) [inline Source/VehiclesCompat/VehiclesCompat/VehicleArmorPatch.cs:25]: invert check to get rid of if else

**Review by trikanyo — DISMISSED** [review]

**Review by N7Huntsman — CHANGES_REQUESTED** [review]
No errors associated with it, but the ammo system options are not appearing on the CE main menu mod settings panel with this assembly. https://i.imgur.com/6X6ggt1.jpg

**Comment by SerinaCE** [comment]
server down

**Review by N7Huntsman — APPROVED** [review]
Everything works in normal, generic, and no ammo modes. The issue with the mod menu panel has been resolved.

**Pull request merged**